registration/app/registration/services/NotificationRegistrarProvider.scala

```diff
@@ -1,15 +1,18 @@
 package registration.services
 
 import error.{NotificationsError, RequestError}
-import models.{Android, iOS, Registration, WindowsMobile}
+import models.{Android, iOS, Platform, Registration, WindowsMobile}
 import registration.services.azure.{APNSNotificationRegistrar, GCMNotificationRegistrar, WindowsNotificationRegistrar}
 
 import scala.concurrent.ExecutionContext
 import cats.data.Xor
 import cats.implicits._
 
 trait RegistrarProvider {
-  def registrarFor(registration: Registration): Xor[NotificationsError, NotificationRegistrar]
+  def registrarFor(registration: Registration): Xor[NotificationsError, NotificationRegistrar] =
+    registrarFor(registration.platform)
+
+  def registrarFor(platform: Platform): Xor[NotificationsError, NotificationRegistrar]
```

Comment:
not really. I'm used to Xor[A, B] rather than fancy syntax, but that's because I'm old school. I'm fine with the other